app/controllers/sessions_controller.rb

```diff
@@ -30,4 +29,16 @@ def delete_user_session!
     reset_session
     session[:user_token] = nil
   end
+
+  def redirect_by_status(user)
+    if user.status == 'active'
+      redirect_to root_url, notice: "Successfully signed in!"
+    elsif user.status == 'pending_email'
+      redirect_to resend_preview_url
+    elsif user.status == 'blocked'
+      redirect_to root_url
+    else
+      render template: "shared/500.html", status: 500
```

Comment:
500 is for internal server error it will be raised if something unexpected happens but if you putting it into code then you expect it, so use another error for that purpose or just remove that part, Rails handling 500/404 automatically.

https://docs.apigee.com/api-platform/troubleshoot/runtime/500-internal-server-error-0

Reply:
There should be some wrong user status logic handling here, What should I do in `else` section?

Reply:
Convention: render as 501 page